Manatee.Json/Schema/Keywords/IRequiresEvaluatedPropretyNamesTracking.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Manatee.Json.Schema
+{
+	/// <summary>
+	/// Can be added to a Json Schema Keyword to specify that it requires Evaluated Property Name tracking.
+	/// </summary>
+	public interface IRequiresEvaluatedPropertyNamesTracking { }
```

Comment:
I think these could be combined into an `IRequireAnnotations` keyword.  That's really what's going on here anyway.  If you read the spec, the keywords on which these depend generate annotations that the `unevaluated*` keywords depend upon.  I just implemented it a bit differently.  (I might have an overhaul at some point to do it the way the spec suggests, but for now, this works.)

Reply:
I'm fine combining them, but I want to double check that you're aware that the broken tests you mentioned in the last PR are broken against `master` too, and not just my changes.

Reply:
Yes I know. It's a separate issue.  That said, it's related to these keywords, and I don't want to fix it until your changes are in. Moreover I think that your changes could help fix it.